delegate: modify SHOW ENUMS and SHOW TYPES to display empty ENUMS

otan opened the pull request:
Fixes #55137 
Based on #55139 

Release note (bug fix): Fix a bug where empty enums did not show up for
SHOW ENUMS or SHOW TYPES.

otan committed:
delegate: fix SHOW ENUMS column names

Release note (bug fix): Fix SHOW ENUMS column names to have `values`
instead of `string_agg` for column names, and `owner` for the owner
itself.

Release note (bug fix): Fix SHOW TYPES to show owner column instead of
the value column.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55143)
<!-- Reviewable:end -->

ajstorm reviewed (commented):
> Reviewed 3 of 3 files at r1, 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55143)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @arulajmani)



<!-- Sent from Reviewable.io -->

ajstorm reviewed (approved):
> Reviewed 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55143)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani)



<!-- Sent from Reviewable.io -->

otan committed:
delegate: modify SHOW ENUMS and SHOW TYPES to display empty ENUMS

Release note (bug fix): Fix a bug where empty enums did not show up for
SHOW ENUMS or SHOW TYPES.

otan commented:
thanks!

bors r=ajstorm
